### PR TITLE
WT-7363 Add support for dumping history store output in hex format

### PR DESCRIPTION
**Pull request by AlexeyAnisimovAU**
Added support for a combination of `-p` and `-x` options to `wt dump` command. In this case, the output will be the same as with `-p` except the raw data arrays. They will be formatted the way hex does that - without leading escape symbols before each byte and without trying to convert each byte to a char. For example this raw bytes dump in pretty format `\0a\1c@\0b\1fG\0c"N\00` will look like this in pretty hex `0a1c400b1f470c224e00`

### CODE REVIEW

**Comment by AlexeyAnisimovAU** [comment]
evergreen retry

File: dist/api_data.py [hunk 1]
```diff
@@ -1432,10 +1432,11 @@ def __ge__(self, other):
         selects a simple hexadecimal format, "json" selects a JSON format
         with each record formatted as fields named by column names if
         available, "pretty" selects a human-readable format (making it
-        incompatible with the "load") and "print" selects a format where only
-        non-printing characters are hexadecimal encoded.  These formats are
-        compatible with the @ref util_dump and @ref util_load commands''',
-        choices=['hex', 'json', 'pretty', 'print']),
+        incompatible with the "load"), "pretty_hex" is similar to "pretty" (also incompatible with
+        "load") except raw byte data elements will be printed like "hex" format does that, and
```
Comment (sueloverso) [inline dist/api_data.py:1436]: Remove 'does that'.

File: src/cursor/cur_std.c [hunk 2]
```diff
@@ -1189,6 +1189,8 @@ __wt_cursor_init(
             dump_flag = WT_CURSTD_DUMP_PRINT;
         else if (WT_STRING_MATCH("pretty", cval.str, cval.len))
             dump_flag = WT_CURSTD_DUMP_PRETTY;
+        else if (WT_STRING_MATCH("pretty_hex", cval.str, cval.len))
+            dump_flag = WT_CURSTD_DUMP_PRETTY | WT_CURSTD_DUMP_HEX;
```
Comment (sueloverso) [inline src/cursor/cur_std.c:1193]: Alphabetize the flags on the line.

File: src/docs/command-line.dox [hunk 3]
```diff
@@ -187,7 +187,8 @@ format.
 
 @par <code>-p</code>
 Dump in human-readable format (pretty-print). The \c -p flag is incompatible
-with the \c load command.
+with the \c load command. Can be combined with \c -x In this case raw data elements will be
```
Comment (sueloverso) [inline src/docs/command-line.dox:190]: `Can be` should be a full complete sentence, ending in a period and adding a subject. `The \c -p flag can be combined with \c -x. In this case...`

File: src/docs/command-line.dox [hunk 4]
```diff
@@ -199,7 +200,9 @@ timestamp.
 
 @par <code>-x</code>
 Dump all characters in a hexadecimal encoding (the default is to leave
-printable characters unencoded).
+printable characters unencoded). Can be combined with \c -p In this case it will format all data
```
Comment (sueloverso) [inline src/docs/command-line.dox:203]: Same here. Don't start the sentence with `Can be`

File: test/suite/test_dump01.py [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+
+import os, shutil
+import wiredtiger, wttest
+
+from suite_subprocess import suite_subprocess
+
+# test_pretty_hex_dump
+#    Utilities: wt dump
+# Test the dump utility with pretty hex flag
+class test_pretty_hex_dump(wttest.WiredTigerTestCase, suite_subprocess):
```
Comment (sueloverso) [inline test/suite/test_dump01.py:41]: Do we need an entirely new test? Can we just add `pretty` and `pretty_hex` as scenarios under `dump_fmt` in `test_dump.py`?

File: test/test_coverage.md [hunk 6]
```diff
@@ -53,4 +53,4 @@
 |Truncate||[test_truncate01.py](../test/suite/test_truncate01.py)
 |Truncate|Prepare|[test_prepare13.py](../test/suite/test_prepare13.py)
 |Verify|Prepare|[test_prepare_hs03.py](../test/suite/test_prepare_hs03.py), [test_timestamp18.py](../test/suite/test_timestamp18.py)
-|Wt Util||[test_backup01.py](../test/suite/test_backup01.py), [test_dump.py](../test/suite/test_dump.py), [test_util11.py](../test/suite/test_util11.py)
+|Wt Util||[test_backup01.py](../test/suite/test_backup01.py), [test_dump.py](../test/suite/test_dump.py), [test_dump01.py](../test/suite/test_dump01.py), [test_util11.py](../test/suite/test_util11.py)
```
Comment (sueloverso) [inline test/test_coverage.md:56]: This change is unnecessary if you add scenarios to `test_dump.py`.

**Review by sueloverso — COMMENTED** [review]
I have a few questions. This mostly looks good.

**Comment by AlexeyAnisimovAU** [comment]
> Do we need an entirely new test? Can we just add `pretty` and `pretty_hex` as scenarios under `dump_fmt` in test_dump.py?

I wanted to do that in the first place. But later I realized that if we add `pretty_hex` test to `test_dump` class, we'll have to skip all permutations but one for `pretty_hex` test and skip `pretty_hex` for existing test cases. At the moment the test has 54 scenarios, when I add one more `dump_fmt` it bumps up the scenarios count to 81. Probably it's more reasonable to have a separate class that does just one thing - tests `pretty_hex` dump.  I thought of adding the class to `test_dump.py` but it already has 2 classes. The trend I see in the majority of python tests is to have one class per file.

**Comment by AlexeyAnisimovAU** [comment]
evergreen retry

File: src/docs/command-line.dox [hunk 7]
```diff
@@ -187,7 +187,8 @@ format.
 
 @par <code>-p</code>
 Dump in human-readable format (pretty-print). The \c -p flag is incompatible
-with the \c load command.
+with the \c load command. The \c -p flag can be combined with \c -x. In this case raw data elements
+will be formatted like \c -x does that.
```
Comment (sueloverso) [inline src/docs/command-line.dox:191]: Please fix all occurrences of `like -x does that` to something ilke "...raw data elements will be formatted like \c -x with hexadecimal encoding". The `does that` is not clear to the reader.

File: src/docs/command-line.dox [hunk 8]
```diff
@@ -199,7 +200,9 @@ timestamp.
 
 @par <code>-x</code>
 Dump all characters in a hexadecimal encoding (the default is to leave
-printable characters unencoded).
+printable characters unencoded). The \c -x flag can be combined with \c -p. In this case it will
+format all data similar to \c -p except raw data elements will be formatted like \c -x does that.
```
Comment (sueloverso) [inline src/docs/command-line.dox:204]: Here too.

File: src/docs/dump-formats.dox [hunk 9]
```diff
@@ -97,6 +99,9 @@ literal character is written as a pair of characters (first the
 high-nibble and then the low-nibble).  For example, "0a" would be an
 ASCII newline character and "1b" would be an ASCII escape character.
 
+The printable hexadecimal format is the same as the printable except all raw
+byte array elements will be printed as in the hexadecimal.
```
Comment (sueloverso) [inline src/docs/dump-formats.dox:103]: Change `will be printed as in the hexadecimal` to `will be printed in hexadecimal format.`

File: src/utilities/util_dump.c [hunk 10]
```diff
@@ -33,12 +34,15 @@ usage(void)
     static const char *options[] = {"-c checkpoint",
       "dump as of the named checkpoint (the default is the most recent version of the data)",
       "-f output", "dump to the specified file (the default is stdout)", "-j",
-      "dump in JSON format", "-p", "dump in human readable format (pretty-print)", "-r",
-      "dump in reverse order", "-t timestamp",
+      "dump in JSON format", "-p",
+      "dump in human readable format (pretty-print). Can be combined with -x In this case raw data "
+      "elements will be formatted like -x does that.",
```
Comment (sueloverso) [inline src/utilities/util_dump.c:39]: Same here. Reword to remove `does that`.

File: src/utilities/util_dump.c [hunk 11]
```diff
@@ -33,12 +34,15 @@ usage(void)
...
+      "-r", "dump in reverse order", "-t timestamp",
       "dump as of the specified timestamp (the default is the most recent version of the data)",
       "-x",
       "dump all characters in a hexadecimal encoding (by default printable characters are not "
-      "encoded)",
+      "encoded). Can be combined with -p In this case it will format all data similar to -p except "
+      "raw data elements will be formatted like -x does that.",
```
Comment (sueloverso) [inline src/utilities/util_dump.c:45]: And here.

File: src/utilities/util_dump.c [hunk 12]
```diff
@@ -559,12 +568,36 @@ dump_table_parts_config(
...
+static const char *
+get_dump_type(bool pretty, bool hex, bool json)
+{
+    const char *result;
+
+    result = NULL;
+
+    if (json)
+        result = "json";
+    else if (pretty && hex)
```
Comment (sueloverso) [inline src/utilities/util_dump.c:583]: Alphabetize `(hex && pretty)`.

**Review by sueloverso — APPROVED** [review]
I still have some minor wording changes I'd like you to make but once those are resolved, this LGTM.

**Review by bvpvamsikrishna — APPROVED** [review]
lgtm

**Comment by AlexeyAnisimovAU** [comment]
Thanks for the review!
Thank you @sueloverso for your valuable comments!

**Pull request merged**